src/main/java/array/MainObject_Advance.java

```diff
@@ -171,9 +238,141 @@ public static void main(String[] args) {
             element.setCurrentBalance(element.reCalculateBalance());
             showInfoBasic(element);
         }
+        printSeparatedLine(60, "~");
+
+        // Câu 12
+        System.out.println("Câu 12: ");
+        for (Account element : listAccount) {
+            showInfo(element, element.getHistories());
+        }
+        printSeparatedLine(60, "~");
+
+        // Câu 13
+        System.out.println("Câu 13: ");
+        int number13 = 0;
+        for (Account element : listAccount) {
+            number13 = element.getHistories().size() + number13;
+        }
+        System.out.println("Tổng số giao dịch của hệ thống là: " + number13);
+        printSeparatedLine(60, "~");
+
+        // Câu 14:
+        System.out.println("Câu 14: ");
+        List<AccountHistory> typeInList = new ArrayList<>();
+        for (Account element : listAccount) {
+            List<AccountHistory> listAccountHistory = element.getAccountByType(AccountHistory.Type.in);
+            for (AccountHistory accountHistory : listAccountHistory) {
+                accountHistory.fullName = element.getFirstName() + " " + element.getLastName();
+                typeInList.add(accountHistory);
+            }
+        }
+        Collections.sort(typeInList, comparing(AccountHistory::getCreatedDate).reversed());
+        for (AccountHistory element : typeInList) {
+            element.showInfoWithoutType();
+        }
+        printSeparatedLine(60, "~");
+
+        // Câu 15:
+        System.out.println("Câu 15: ");
+        long totalInBalance = 0;
+        for (AccountHistory element : typeInList) {
+            totalInBalance = element.getBalance() + totalInBalance;
+        }
+        System.out.println("TỔNG số tiền đã được NẠP VÀO hệ thống = " + totalInBalance);
+        printSeparatedLine(60, "~");
+
+        // Câu 16:
+        System.out.println("Câu 16: ");
+        long totalOutBalance = 0;
+        for (Account element : listAccount) {
+            List<AccountHistory> cau16 = element.getAccountByType(AccountHistory.Type.out);
+            for (AccountHistory balance : cau16) {
+                totalOutBalance = balance.getBalance() + totalOutBalance;
+            }
+
+        }
+        System.out.println("TỔNG số tiền đã được RÚT RA khỏi hệ thống = " + totalOutBalance);
+        printSeparatedLine(60, "~");
+
+        //   Câu 17:
```

Comment:
Câu này approach có vấn đề à e. Cái a cần là lấy những giao dịch có số tiền lớn nhất trong tất cả cách giao dịch của mọi tài khoản. Nên line 309 mà mình đang sort ở trong for là chắc chắn chưa đúng rồi e à.